WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -2137,68 +2141,12 @@ private extension EditableOrderViewModel {
-                        .map { ProductVariationAttribute(id: $0.metaID, name: $0.name, option: $0.value) }
-                    let productOrVariation: BundledProductConfiguration.ProductOrVariation = existingOrderItem.variationID == 0 ?
-                        .product(id: existingOrderItem.productID): .variation(productID: existingOrderItem.productID,
-                                                                              variationID: existingOrderItem.variationID,
-                                                                              attributes: attributes)
-                    // The quantity per bundle: as a buggy behavior in Pe5pgL-3Vd-p2#quantity-of-bundle-child-order-items, the child item quantity
-                    // can either by multiplied by the bundle quantity or not. To encounter for the edge case, the quantity is only divided by
-                    // the bundle quantity if the child item has at least the same quantity as the bundle.
-                    let quantity = existingOrderItem.quantity >= item.quantity ?
-                    existingOrderItem.quantity * 1.0 / item.quantity: existingOrderItem.quantity
-                    return .init(bundledItemID: bundleItem.bundledItemID,
-                                 productOrVariation: productOrVariation,
-                                 quantity: quantity,
-                                 isOptionalAndSelected: bundleItem.isOptional ? true: nil)
-                }
-            return OrderSyncProductInput(id: item.itemID,
-                                         product: product,
-                                         quantity: quantity,
-                                         discount: discount ?? currentDiscount(on: item),
-                                         baseSubtotal: baseSubtotal(on: item),
-                                         bundleConfiguration: bundleConfiguration)
-        }
-
-        // Return a new input with the new quantity but with the same item id to properly reference the update.
-        return OrderSyncProductInput(id: item.itemID,
-                                     product: product,
-                                     quantity: quantity,
-                                     discount: discount ?? currentDiscount(on: item),
-                                     baseSubtotal: baseSubtotal(on: item),
-                                     bundleConfiguration: bundleConfiguration)
+        ProductInputTransformer.createUpdateProductInput(item: item,
+                                                         childItems: childItems,
+                                                         quantity: quantity,
+                                                         allProducts: Array(allProducts),
+                                                         allProductVariations: allProductVariations,
+                                                         defaultDiscount: currentDiscount(on: item))
```

Comment:
❓ Two parameters seem to be missing:

```suggestion
        ProductInputTransformer.createUpdateProductInput(item: item,
                                                         childItems: childItems,
                                                         quantity: quantity,
                                                         discount: discount,
                                                         bundleConfiguration: bundleConfiguration,
                                                         allProducts: Array(allProducts),
                                                         allProductVariations: allProductVariations,
                                                         defaultDiscount: currentDiscount(on: item))
```

Reply:
Good catch 😅